test/built-ins/ThrowTypeError/name.js

```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2016 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-%throwtypeerror%
+description: >
+  %ThrowTypeError% is an anonymous function.
+info: >
+  %ThrowTypeError% ( )
+
+  The %ThrowTypeError% intrinsic is an anonymous built-in function
+  object that is defined once for each Realm.
+---*/
+
+var ThrowTypeError = Object.getOwnPropertyDescriptor(function(){ "use strict"; return arguments; }(), "callee").get;
+
+assert.sameValue(Object.prototype.hasOwnProperty.call(ThrowTypeError, "name"), false);
```

Comment:
Isn't this affected bythe var binding?

https://tc39.github.io/ecma262/#sec-variable-statement-runtime-semantics-evaluation

```
VariableDeclaration: BindingIdentifier Initializer 

  1.Let bindingId be StringValue of BindingIdentifier.
  2.Let lhs be ? ResolveBinding(bindingId).
  3.Let rhs be the result of evaluating Initializer.
  4.Let value be ? GetValue(rhs).
  5.If IsAnonymousFunctionDefinition(Initializer) is true, then
    a.Let hasNameProperty be ? HasOwnProperty(value, "name").
    b.If hasNameProperty is false, perform SetFunctionName(value, bindingId).
```